Create vecs for indices and vertices in tutorial

dariuswiles opened the pull request.

KyleMayes commented:
This is unnecessary due to `AppData` implementing `Default` and being constructed using `AppData::default()`. However, if you found this confusing/unexpected it might be good to call out that these `Vec`s are already initialized.

dariuswiles commented:
I think I screwed my code up as I was copying snippets from the tutorial, and during my efforts to appease the compiler I added these lines and made other changes until everything worked. You're right that these lines are redundant and I've removed them from my code and, not surprisingly, everything still works. Sorry for wasting your time.

KyleMayes commented:
No problem, this PR and #208 are still good feedback.
Thanks for taking the time to submit issues/PRs!
